sklearn/neighbors/_graph.py

```diff
@@ -36,6 +43,20 @@ def _query_include_self(X, include_self, mode):
     return X
 
 
+@validate_params(
+    {
+        "X": ["array-like", BallTree, KNeighborsMixin],
+        "n_neighbors": [Interval(Integral, 1, None, closed="left")],
+        "mode": [StrOptions({"connectivity", "distance"})],
+        "metric": "no_validation",  # metric is passed without param validation
+        # here, because VALID_METRICS are extracted and defined in
+        # :class:`~sklearn.neighbors._base.NeighborsBase`
```

Comment:
I think that I would prefer to check again. While having a `NeighborsBase` means that the parameter will be validated, this is not the case with a `BallTree`. So I think this is safer to just make the check


```suggestion
        "metric": [StrOptions(set(itertools.chain(*VALID_METRICS.values()))), callable],
```